cmd/download.go

```diff
@@ -264,6 +197,119 @@ func (d download) needsSlugWhenGivenTrackOrTeam() error {
 	return nil
 }
 
+func (d download) metadata() *ws.ExerciseMetadata {
```

Comment:
I think this method might be going beyond the scope of the download type. Which If I understand correctly is a custom type used for parsing flags and setting up the download command. Having actual exercise metadata and solutionFile data :arrow_down: tied to it doesn't seem correct.